com.unity.render-pipelines.high-definition/Runtime/Lighting/Shadow/HDShadowAlgorithms.hlsl

```diff
@@ -1,31 +1,10 @@
-// Various shadow algorithms
-// There are two variants provided, one takes the texture and sampler explicitly so they can be statically passed in.
-// The variant without resource parameters dynamically accesses the texture when sampling.
-
-// WARNINGS:
-// Keep in sync with HDShadowManager::GetDirectionalShadowAlgorithm()
-// Be careful this require to update GetPunctualFilterWidthInTexels() in C# as well!
+// Configure which shadow algorithms to use per shadow level quality
 
 // Since we use slope-scale bias, the constant bias is for now set as a small fixed value
 #define FIXED_UNIFORM_BIAS (1.0f / 65536.0f)
 
-#if (defined(_ENABLE_SHADOW_MATTE) && SHADERPASS == SHADERPASS_FORWARD_UNLIT)
```

Comment:
How does that work for shadow matte now? Unlit does not go through deferred.compute and I don't think it has a multi compile for shadows.

Reply:
See change in shader graph. Shadow matte generated from shader graph correctly generate a multi compile for shadow.  (there is no shadow matte from unlit.shader, only from SG unlit).